### PR TITLE
feat: Implement ConnectedAccountsMenu buttons

### PR DESCRIPTION
**Pull request by jonybur**
## **Description**

Implements the 'Switch to this account' and the 'Disconnect' buttons on the ConnectedAccountsMenu panel.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/23258?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/2172

## **Manual testing steps**

1. Connect two accounts to a dApp
2. Open the Connected Accounts menu panel
3. Switch between the accounts
4. Disconnect an account

## **Screenshots/Recordings**

https://github.com/MetaMask/metamask-extension/assets/11148144/eca77143-c53b-4978-b698-3cb4d2e3ac95

https://github.com/MetaMask/metamask-extension/assets/11148144/13d2a4bf-042f-477a-85b6-3459bbc0eb39

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/components/multichain/connected-accounts-menu/index.scss [hunk 1]
```diff
@@ -0,0 +1,8 @@
+@use "design-system";
```
Comment (NidhiKJha) [inline ui/components/multichain/connected-accounts-menu/index.scss:1]: Do we need this file?

File: ui/components/multichain/connected-accounts-menu/connected-accounts-menu.tsx [hunk 2]
```diff
@@ -15,18 +16,45 @@ import {
...
+import {
+  MetaMetricsEventCategory,
+  MetaMetricsEventName,
+} from '../../../../shared/constants/metametrics';
+import { MetaMetricsContext } from '../../../contexts/metametrics';
+import { getOriginOfCurrentTab } from '../../../selectors';
 
 const TsMenuItem = MenuItem as any;
 
+interface Identity {
```
Comment (NidhiKJha) [inline ui/components/multichain/connected-accounts-menu/connected-accounts-menu.tsx:32]: Let's keep types in a separate file

File: ui/components/multichain/connected-accounts-menu/connected-accounts-menu.tsx [hunk 3]
```diff
@@ -67,13 +95,30 @@ export const ConnectedAccountsMenu = ({
           <TsMenuItem
             iconName={IconName.SwapHorizontal}
             data-testid="switch-account-menu-item"
+            onClick={() => {
+              trackEvent({
+                category: MetaMetricsEventCategory.Navigation,
+                event: MetaMetricsEventName.NavAccountSwitched,
+                properties: {
+                  location: 'Main Menu',
```
Comment (NidhiKJha) [inline ui/components/multichain/connected-accounts-menu/connected-accounts-menu.tsx:103]: So, metametrics are specific to component. I don't think we should use these here. We might create new one. Not a part of this ticket

File: ui/components/multichain/connected-accounts-menu/connected-accounts-menu.tsx [hunk 4]
```diff
@@ -67,13 +95,30 @@ export const ConnectedAccountsMenu = ({
...
+                },
+              });
+              dispatch(setSelectedAccount(identity.address));
+              onClose();
+              closeMenu();
```
Comment (NidhiKJha) [inline ui/components/multichain/connected-accounts-menu/connected-accounts-menu.tsx:108]: Let's change it to this
```suggestion
              dispatch(setSelectedAccount(identity.address));
              onClose();
              closeMenu();
```

**Review by NidhiKJha — CHANGES_REQUESTED** [review]
Great work @jonybur Left some suggestions but overall looks great!
Can we add a condition to not show switch to this account option when there is only one account connected?
Screenshot 2024-03-04 at 7 21 17 PM

File: ui/components/multichain/connected-accounts-menu/connected-accounts-menu.tsx (hunk 2 above)
Comment (jonybur) [reply ui/components/multichain/connected-accounts-menu/connected-accounts-menu.tsx:32]: Done

File: ui/components/multichain/connected-accounts-menu/connected-accounts-menu.test.tsx [hunk 5]
```diff
@@ -1,59 +1,65 @@
...
+    const { getByTestId } = renderComponent();
+    expect(getByTestId('permission-details-menu-item')).toBeInTheDocument();
   });
 
-  it('renders switch to this account menu item', async () => {
-    await act(async () => {
-      const { getByText } = render();
-      expect(getByText('Switch to this account')).toBeInTheDocument();
-    });
+  it('renders switch to this account menu item if account switcher is enabled', async () => {
```
Comment (NidhiKJha) [inline ui/components/multichain/connected-accounts-menu/connected-accounts-menu.test.tsx:43]: Really nice!

File: ui/components/multichain/connected-accounts-menu/connected-accounts-menu.tsx [hunk 6]
```diff
@@ -15,18 +16,32 @@ import {
...
+import { getOriginOfCurrentTab } from '../../../selectors';
+import { Identity } from './connected-accounts-menu.types';
 
 const TsMenuItem = MenuItem as any;
 
 export const ConnectedAccountsMenu = ({
   isOpen,
+  identity,
   anchorElement,
+  disableAccountSwitcher,
```
Comment (NidhiKJha) [inline ui/components/multichain/connected-accounts-menu/connected-accounts-menu.tsx:32]: By default this should be false?

File: ui/components/multichain/account-list-item/account-list-item.js [hunk 7]
```diff
@@ -355,6 +357,16 @@ export const AccountListItem = ({
           isConnected={isConnected}
         />
       )}
+      {menuType === AccountListItemMenuTypes.Connection && (
+        <ConnectedAccountsMenu
+          anchorElement={accountListItemMenuElement}
+          identity={identity}
+          onClose={() => setAccountOptionsMenuOpen(false)}
+          closeMenu={closeMenu}
+          disableAccountSwitcher={accountsCount === 1}
```
Comment (NidhiKJha) [inline ui/components/multichain/account-list-item/account-list-item.js:366]: Hmmm...Do you think we should create a separate variable for this?

**Review by NidhiKJha — CHANGES_REQUESTED** [review]
Almost there! Just some small nits

File: ui/components/multichain/connected-accounts-menu/connected-accounts-menu.test.tsx (hunk 5 above)
Comment (jonybur) [reply ui/components/multichain/connected-accounts-menu/connected-accounts-menu.test.tsx:43]: Thanks!

**Review by NidhiKJha — DISMISSED** [review]
LGTM

**Review by darkwing — DISMISSED** [review]

**Review by darkwing — APPROVED** [review]

**Review by DDDDDanica — APPROVED** [review]

**Pull request merged**